### PR TITLE
refactor : rm mod uses new profile impl.

### PR DESCRIPTION
**Pull request by UsairimIsani**
# RM - Migrated to new Profile Impl.

@rusty-snake

### CODE REVIEW

File: src/rm.rs [hunk 1]
```diff
@@ -27,9 +27,12 @@ pub fn start(cli: &ArgMatches<'_>) {
...
     for profile in profiles {
-        let profile = get_name1(profile);
-        trace!("Deleting '{}'.", profile);
-        remove_file(profile_path!(USER / &profile))
-            .unwrap_or_else(|err| error!("Failed to delete '{}': {}", profile, err));
+        let profile = Profile::new(
+            profile,
+            ProfileFlags::LOOKUP_USER | ProfileFlags::DENY_BY_PATH | ProfileFlags::ASSUME_EXISTENCE,
+        )
+        .unwrap();
```
Comment (rusty-snake) [inline src/rm.rs:34]:
```suggestion
        .unwrap();
        trace!("Deleting '{}'.", profile.full_name());
```

**Review by rusty-snake — APPROVED** [review]
LGTM

**Comment by rusty-snake** [comment]
Thanks for this contribution!

**Comment by UsairimIsani** [comment]
Want to do more but... looking for work is exhausting.

**Comment by rusty-snake** [comment]
Sounds great, here are some suggestions:

**tests**
- Unit tests for utils.rs, profile.rs, macros::fjp_version, location.rs
- Integration tests (possible crates: assert-cmd & assert-fs) e.g. works `fjp disable firefox`, ...

**new subcommands**
- trash, mv, cp (see TODO.md)
- Implementing other subcommands in TODO.md which handle profile content should wait until profile_stream.rs is stable (actually it is experimental).

**fine tuning**
- As you know :wink: the new profile sys.
- See edit and generate-standalone in TODO.md

**docs**
- Rewrite the manpage using roff (https://www.cyberciti.biz/faq/linux-unix-creating-a-manpage/)
- More long_help in cli.yaml

**misc**
- Improvements bash/zsh completions.
- Scripts to build fjp.deb/fjp.pkg.tar.xz packages (platform/{arch,debian}).

**Pull request merged**